runtime/src/launch/mpirun/launch-mpirun.c

```diff
@@ -52,10 +52,53 @@ static char** chpl_launch_create_argv(const char *launch_cmd,
   }
 
 
+  int largc = 5;
   largv[0] = (char *) launch_cmd;
   largv[1] = (char *) "-np";
   sprintf(_nlbuf, "%d", numranks);
   largv[2] = _nlbuf;
+  largv[3] = (char*) "-E";
+  largv[4] = chpl_get_enviro_keys(',');
+
+  // If any of the relevant character set environment variables
+  // are set, replicate the state of all of them.  This needs to
+  // be done separately from the -E mechanism because the
+  // launcher is written in Perl, which mangles the character
```

Comment:
This calls `mpirun`, not a gasnet wrapper. How sure are you that mpirun may be written in perl?

Reply:
`third-party/gasnet/gasnet-src/other/mpirun_common.in` seems to imply that mpirun is a wrapper for the Perl-based gasnet_mpi.

Reply:
I'm not following. We're _not_ calling a gasnet wrapper. We're literally calling the system `mpirun`

gasnetrun_* launchers that use mpi-spawning may end up calling gasnetrun_mpi, but that's not related to this. Under the covers gasnetrun_mpi will call the system `mpirun`, but again not related to this.

Reply:
OK, in that case, I can remove that file from my changes.